flocker/node/agents/blockdevice.py

```diff
@@ -220,6 +440,21 @@ def attach_volume(blockdevice_id, host):
             ``host``.
         """
 
+    def detach_volume(blockdevice_id):
+        """
+        Detach ``blockdevice_id`` from whatever host it is attached to.
+
+        :param unicode blockdevice_id: The unique identifier for the block
+            device being detached.
+
+        :raises UnknownVolume: If the supplied ``blockdevice_id`` does not
+            exist.
+        :raises UnattachedVolume: If the supplied ``blockdevice_id`` is
+            not attached to anything.
+        :returns: A ``BlockDeviceVolume`` with a ``host`` attribute set to
```

Comment:
This return result means every backend implementation has to implement the same in-memory generic functionality. Seems better to have the caller that cares do it, once, and make life easier for backend implemetnations.